x-pack/plugins/encrypted_saved_objects/server/create_migration.test.ts

```diff
@@ -111,6 +112,402 @@ describe('createMigration()', () => {
+        throw new EncryptionError(
+          `Unable to decrypt attribute "'attribute'"`,
+          'attribute',
+          EncryptionErrorOperation.Decryption,
+          new Error('decryption failed')
+        );
+      });
+
+      encryptionSavedObjectService.stripOrDecryptAttributesSync.mockReturnValueOnce({
+        attributes: strippedAttributes,
+      });
+
+      noopMigration(
+        {
+          id: '123',
+          type: 'known-type-1',
+          namespace: 'namespace',
+          attributes,
+        },
+        migrationContext
+      );
+
+      expect(encryptionSavedObjectService.stripOrDecryptAttributesSync).toHaveBeenCalledWith(
+        {
+          id: '123',
+          type: 'known-type-1',
+          namespace: 'namespace',
+        },
+        attributes,
+        { convertToMultiNamespaceType: false }
+      );
+
+      expect(migrationFunc).toHaveBeenCalled();
+      expect(encryptionSavedObjectService.encryptAttributesSync).toHaveBeenCalledWith(
+        {
+          id: '123',
+          type: 'known-type-1',
+          namespace: 'namespace',
+        },
+        strippedAttributes
+      );
+    });
+
+    it('throws error on migration failure', () => {
+      const instantiateServiceWithLegacyType = jest.fn(() =>
+        encryptedSavedObjectsServiceMock.create()
+      );
+      const migrationFunc = jest.fn((doc) => {
+        throw new Error('migration failed!');
+      });
+
+      const migrationCreator = getCreateMigration(
+        encryptionSavedObjectService,
+        instantiateServiceWithLegacyType
+      );
+      const noopMigration = migrationCreator<InputType, MigrationType>({
+        isMigrationNeededPredicate(doc): doc is SavedObjectUnsanitizedDoc<InputType> {
+          return true;
+        },
+        migration: migrationFunc,
+      });
+
+      const attributes = {
+        firstAttr: 'first_attr',
+      };
+
+      encryptionSavedObjectService.decryptAttributesSync.mockReturnValueOnce(attributes);
+
+      expect(() => {
+        noopMigration(
+          {
+            id: '123',
+            type: 'known-type-1',
+            namespace: 'namespace',
+            attributes,
+          },
+          migrationContext
+        );
+      }).toThrowError(`migration failed!`);
+
+      expect(encryptionSavedObjectService.decryptAttributesSync).toHaveBeenCalledWith(
+        {
+          id: '123',
+          type: 'known-type-1',
+          namespace: 'namespace',
+        },
+        attributes,
+        { convertToMultiNamespaceType: false }
+      );
+
+      expect(migrationFunc).toHaveBeenCalled();
+      expect(encryptionSavedObjectService.encryptAttributesSync).not.toHaveBeenCalled();
+    });
+
+    it('throws error on migration failure even if shouldMigrateIfDecryptionFails is true', () => {
+      const instantiateServiceWithLegacyType = jest.fn(() =>
+        encryptedSavedObjectsServiceMock.create()
+      );
+      const migrationFunc = jest.fn((doc) => {
```

Comment:
__optional nit:__
```suggestion
      const migrationFunc = jest.fn(() => {
```

Reply:
Updated in https://github.com/elastic/kibana/pull/105968/commits/1365d05c60aeae8c4bd64962cfa6f93293dc4e0d